alicloud/resource_alicloud_container_cluster.go

```diff
@@ -0,0 +1,253 @@
+		args.VSwitchID = v.(string)
+		args.SubnetCIDR = cidr.(string)
+
+		vswInfo, _, err := client.vpcconn.DescribeVSwitches(&ecs.DescribeVSwitchesArgs{
+			RegionId:  getRegion(d, meta),
+			VSwitchId: v.(string),
+		})
+		if err != nil {
+			return fmt.Errorf("Error DescribeVSwitches: %#v", err)
+		}
+		if len(vswInfo) < 1 {
+			return fmt.Errorf("There is not found specified vswitch: %s, please check and try again.", v.(string))
+		}
+		args.VPCID = vswInfo[0].VpcId
+	} else {
+		args.NetworkMode = cs.ClassicNetwork
+	}
+
+	if imageId, ok := d.GetOk("image_id"); ok {
+		connection := client.ecsconn
+		argsImage := &ecs.DescribeImagesArgs{
+			RegionId: getRegion(d, meta),
+			ImageId:  imageId.(string),
+		}
+		if _, _, err := connection.DescribeImages(argsImage); err != nil {
+			return err
+		}
+
+		args.ECSImageID = imageId.(string)
+	}
+
+	region := getRegion(d, meta)
+	cluster, err := conn.CreateCluster(region, args)
+
+	if err != nil {
+		return fmt.Errorf("Creating container Cluster got an error: %#v", err)
+	}
+
+	err = conn.WaitForClusterAsyn(cluster.ClusterID, cs.Running, 500)
+
+	if err != nil {
+		return fmt.Errorf("Waitting for container Cluster %#v got an error: %#v", cs.Running, err)
+	}
+
+	d.SetId(cluster.ClusterID)
+
+	return resourceAlicloudContainerClusterUpdate(d, meta)
+}
+
+func resourceAlicloudContainerClusterUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AliyunClient).csconn
+	d.Partial(true)
+	if d.HasChange("size") && !d.IsNewResource() {
+		o, n := d.GetChange("size")
+		oi := o.(int)
+		ni := n.(int)
+		if ni <= oi {
+			return fmt.Errorf("The new size of clusters must greater than the current. The cluster's current size is %d.", oi)
+		}
+		d.SetPartial("size")
+		err := conn.ResizeCluster(d.Id(), &cs.ClusterResizeArgs{
+			Size:             int64(ni),
+			InstanceType:     d.Get("instance_type").(string),
+			Password:         d.Get("password").(string),
+			DataDiskCategory: ecs.DiskCategory(d.Get("disk_category").(string)),
+			DataDiskSize:     int64(d.Get("disk_size").(int)),
+			ECSImageID:       d.Get("image_id").(string),
+			IOOptimized:      ecs.IoOptimized("true"),
+		})
+		if err != nil {
+			return fmt.Errorf("Resize Cluster got an error: %#v", err)
+		}
+
+		err = conn.WaitForClusterAsyn(d.Id(), cs.Running, 500)
+
+		if err != nil {
+			return fmt.Errorf("Waitting for container Cluster %#v got an error: %#v", cs.Running, err)
+		}
+	}
+	d.Partial(false)
+
+	return resourceAlicloudContainerClusterRead(d, meta)
+}
+
+func resourceAlicloudContainerClusterRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AliyunClient).csconn
+
+	cluster, err := conn.DescribeCluster(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", cluster.Name)
+	d.Set("size", cluster.Size)
+	d.Set("network_mode", cluster.NetworkMode)
+	if cluster.VPCID != "" {
+		d.Set("vpc_id", cluster.VPCID)
+	}
+	if cluster.VSwitchID != "" {
+		d.Set("vswitch_id", cluster.VSwitchID)
+	}
+
```

Comment:
Hi @catsby 

Thanks for your review. The API doesn't return above fields and if them have been changed, the resource would be re-create. I can get their value from schema and return. Could please provide better advice? Thanks.

Regards,
Guimin